src/Microsoft.DotNet.ProjectModel.Workspace/ProjectJsonWorkspace.cs

```diff
@@ -0,0 +1,234 @@
+{
+    public class ProjectJsonWorkspace : Workspace
+    {
+        private Dictionary<string, AssemblyMetadata> _cache = new Dictionary<string, AssemblyMetadata>();
+
+        private readonly string[] _projectPaths;
+
+        public ProjectJsonWorkspace(string projectPath) : this(new[] { projectPath })
+        {
+        }
+
+        public ProjectJsonWorkspace(string[] projectPaths) : base(MefHostServices.DefaultHost, "Custom")
+        {
+            _projectPaths = projectPaths;
+
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            foreach (var projectPath in _projectPaths)
+            {
+                AddProject(projectPath);
+            }
+        }
+
+        private void AddProject(string projectPath)
+        {
+            // Get all of the specific projects (there is a project per framework)
+            foreach (var p in ProjectContext.CreateContextForEachFramework(projectPath))
+            {
+                AddProject(p);
+            }
+        }
+
+        private ProjectId AddProject(ProjectContext project)
+        {
+            // Create the framework specific project and add it to the workspace
+            var projectInfo = ProjectInfo.Create(
+                                ProjectId.CreateNewId(),
+                                VersionStamp.Create(),
+                                project.ProjectFile.Name + "+" + project.TargetFramework,
+                                project.ProjectFile.Name,
+                                LanguageNames.CSharp,
+                                project.ProjectFile.ProjectFilePath);
+
+            OnProjectAdded(projectInfo);
+
+            // TODO: ctor argument?
+            var configuration = "Debug";
+
+            var compilationOptions = project.ProjectFile.GetCompilerOptions(project.TargetFramework, configuration);
+
+            var compilationSettings = ToCompilationSettings(compilationOptions, project.TargetFramework, project.ProjectFile.ProjectDirectory);
+
+            OnParseOptionsChanged(projectInfo.Id, new CSharpParseOptions(compilationSettings.LanguageVersion, preprocessorSymbols: compilationSettings.Defines));
+
+            OnCompilationOptionsChanged(projectInfo.Id, compilationSettings.CompilationOptions);
+
+            foreach (var file in project.ProjectFile.Files.SourceFiles)
+            {
+                AddSourceFile(projectInfo, file);
+            }
+
+            var exporter = project.CreateExporter(configuration);
+
+            foreach (var dependency in exporter.GetDependencies())
+            {
+                var projectDependency = dependency.Library as ProjectDescription;
+                if (projectDependency != null)
+                {
+                    var projectDependencyContext = ProjectContext.Create(projectDependency.Project.ProjectFilePath, projectDependency.Framework);
+
+                    var id = AddProject(projectDependencyContext);
```

Comment:
Need to check that. Probably need to do more caching. @troydai is looking into that.